src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -249,8 +249,17 @@ private void finish() {
         /*
          * If the plugin has been disabled,
          * do not schedule the next execution.
+         *
+         * Additionally, shutdown the async executor
+         * if that has not already been done so.
+         * We do not this TickerTask instance to be
+         * reused following plugin shutdown. Therefore,
+         * we shutdown the ExecutorService.
          */
         if (!plugin.isEnabled()) {
+            if (!asyncExecutor.isShutdown()) {
+                asyncExecutor.shutdown();
```

Comment:
We really should be shutting this down in onDisable

Reply:
So an explicit call to a  `TickerTask#shutdownAsyncTasks` in `Plugin#onDisable`

Reply:
Yep.

Reply:
We should have a stop method which cleans up and stops everything. So, the executor gets shut down, pending tasks get cleared and the queues get processed and cleared.

Reply:
Got it.